src/Features/CSharp/Portable/ExtractMethod/CSharpMethodExtractor.Analyzer.cs

```diff
@@ -35,8 +35,17 @@ protected override bool IsInPrimaryConstructorBaseType()
                     : info.ConvertedType;
             }
 
-            protected override bool ContainsReturnStatementInSelectedCode(ImmutableArray<SyntaxNode> exitPoints)
-                => exitPoints.Any(n => n is ReturnStatementSyntax);
+            protected override ExtractMethodFlowControlInformation GetStatementFlowControlInformation(
+                ControlFlowAnalysis controlFlowAnalysis)
+            {
+                return ExtractMethodFlowControlInformation.Create(
+                    this.SemanticModel.Compilation,
+                    supportsComplexFlowControl: true,
+                    breakStatementCount: controlFlowAnalysis.ExitPoints.Count(n => n is BreakStatementSyntax),
+                    continueStatementCount: controlFlowAnalysis.ExitPoints.Count(n => n is ContinueStatementSyntax),
+                    returnStatementCount: controlFlowAnalysis.ExitPoints.Count(n => n is ReturnStatementSyntax),
+                    endPointIsReachable: controlFlowAnalysis.EndPointIsReachable);
```

Comment:
could make this a simple foreach loop.  but i didn't think it woudl  be a hotspot to warrant it.